libs/vm-modules/include/vm_modules/math/tensor.hpp

```diff
@@ -45,6 +46,57 @@ class VMTensor : public fetch::vm::Object
   using DataType   = fetch::vm_modules::math::DataType;
   using TensorType = typename fetch::math::Tensor<DataType>;
 
+  class TensorEstimator
+  {
+  public:
+    using VMObjectType = VMTensor;
+    using ChargeAmount = fetch::vm::ChargeAmount;
+
+    explicit TensorEstimator(VMObjectType &tensor);
+    ~TensorEstimator() = default;
+
+    ChargeAmount size();
+
+    ChargeAmount AtOne(TensorType::SizeType idx1);
+
+    ChargeAmount AtTwo(uint64_t idx1, uint64_t idx2);
+
+    ChargeAmount AtThree(uint64_t idx1, uint64_t idx2, uint64_t idx3);
+
+    ChargeAmount AtFour(uint64_t idx1, uint64_t idx2, uint64_t idx3, uint64_t idx4);
+
+    ChargeAmount SetAtOne(uint64_t idx1, DataType const &value);
```

Comment:
tensor actually binds up to 6D At and Set methods I believe. so we should decide whether to assign those infinite charge or implement the relevant estimator functions.

Reply:
yes, this is true with Vladimir recent changes

Reply:
Only up to 4 are bound to etch 
https://github.com/MissingNO57/ledger/blob/feature/vm_model_benchmarks/libs/vm-modules/src/math/tensor.cpp#L76